spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DistriParameterSynchronizer.scala

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils
+
+import java.nio.ByteBuffer
+import java.util.concurrent._
+
+import com.intel.analytics.bigdl.parameters.{CompressedTensor, FP16CompressedTensor, SerializerInstance}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import org.apache.commons.lang.exception.ExceptionUtils
+import org.apache.log4j.Logger
+import org.apache.spark.sparkExtension.SparkExtension
+import org.apache.spark.storage.{BlockId, BlockManagerWrapper, StorageLevel}
+
+import scala.collection.mutable
+import scala.reflect.{ClassTag, classTag}
+import scala.collection.JavaConverters._
+
+trait DistriParameterSynchronizer[T] {
+
+  /**
+   * Init synchronization context for new parameter
+   * @param name  identifier for parameter
+   * @param globalSize toal size of parameter
+   * @param priority priority for this parameter
+   */
+  def init(name: String, globalSize: Int, priority: Int = 1): Unit
```

Comment:
what does the globalSize mean? Say the tensor size is 1M, there're 10 nodes. GlobalSize = 10M or 1M?

Reply:
the global size should be 1m in this case, each node handles 1k